executor/adapter.go

```diff
@@ -289,7 +289,7 @@ func (a *ExecStmt) handleNoDelayExecutor(ctx context.Context, sctx sessionctx.Co
 		a.logSlowQuery(txnTS, err == nil)
 	}()
 
-	if sctx.GetSessionVars().EnableChunk && e.supportChunk() {
+	if sctx.GetSessionVars().EnableChunk {
```

Comment:
Is this all true?

Reply:
yes, all executors have supported chunk.

Reply:
So we can remove it.

Reply:
`recordSet.SupportChunk` can not be removed now, it implements `ast.RecordSet` interface, some other implementors of `ast.RecordSet` have not support chunk, we can not remove it by now, see https://internal.pingcap.net/jira/browse/TIDB-117 for more detail.

`Executor.supportChunk` can be safely removed because all executors have supported chunk.

Reply:
I mean EnableChunk. Since it's all true.

Reply:
It will be deleted soon, but before we removing it, we should evaluate the impact of this operation.